lib/http-api.js

```diff
@@ -45,6 +45,7 @@ module.exports.PREFIX_IDENTITY_V1 = "/_matrix/identity/api/v1";
 /**
  * Construct a MatrixHttpApi.
  * @constructor
+ * @param {MatrixClient} client The matrix client instance to use.
```

Comment:
can we call make this an `EventEmitter` rather than a client, so that we aren't tempted to start adding all sorts of binding between the HttpApi and MatrixClient?